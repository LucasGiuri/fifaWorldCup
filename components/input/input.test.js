import React from 'react';

import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Input from './input.tsx';

describe('Input', () => {
  const onChange = jest.fn();

  it('basic snapshot renders', () => {
    const b = renderer.create(<Input type='penalties' onChange={onChange} />);
    const tree = b.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders with input', () => {
    const b = renderer.create(<Input type='penalties' value='fakeValue' onChange={onChange} />);
    const tree = b.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders with isDisabled', () => {
    const b = renderer.create(<Input type='penalties' isDisabled onChange={onChange} />);
    const tree = b.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders with isDisabled and value', () => {
    const b = renderer.create(<Input type='penalties' value='fakeValue' isDisabled onChange={onChange} />);
    const tree = b.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
