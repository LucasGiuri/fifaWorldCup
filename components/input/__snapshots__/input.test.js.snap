// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input basic snapshot renders 1`] = `
.c0 {
  width: 35px;
  height: 35px;
  text-align: center;
  border: 1px solid #696969;
  box-shadow: none;
  border-radius: 50%;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0:focus {
  outline: none;
}

<input
  className="c0"
  min="0"
  onChange={[MockFunction]}
  type="number"
/>
`;

exports[`Input snapshot renders with input 1`] = `
.c0 {
  width: 35px;
  height: 35px;
  text-align: center;
  border: 1px solid #696969;
  box-shadow: none;
  border-radius: 50%;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0:focus {
  outline: none;
}

<input
  className="c0"
  min="0"
  onChange={[MockFunction]}
  type="number"
  value="fakeValue"
/>
`;

exports[`Input snapshot renders with isDisabled 1`] = `
.c0 {
  width: 35px;
  height: 35px;
  text-align: center;
  border: 1px solid #696969;
  box-shadow: none;
  border-radius: 50%;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0:focus {
  outline: none;
}

<input
  className="c0"
  disabled={true}
  min="0"
  onChange={[MockFunction]}
  type="number"
/>
`;

exports[`Input snapshot renders with isDisabled and value 1`] = `
.c0 {
  width: 35px;
  height: 35px;
  text-align: center;
  border: 1px solid #696969;
  box-shadow: none;
  border-radius: 50%;
}

.c0::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0:focus {
  outline: none;
}

<input
  className="c0"
  disabled={true}
  min="0"
  onChange={[MockFunction]}
  type="number"
  value="fakeValue"
/>
`;
